docs/src/environments.md

```diff
@@ -107,3 +112,146 @@ Otherwise, it will resolve the latest versions of the dependencies compatible wi
+
+julia> using BenchmarkTools
+
+```
+
+Loading `BenchmarkTools` in MyProject environment works even if this package is not recorded in `~/MyProject/Project.toml`. This is
+because Julia maintains a environments stack via the `LOAD_PATH` variable, see also the [environment stacks](https://docs.julialang.org/en/v1/manual/code-loading/#Environment-stacks) section in the julia manual.
+
+```julia-repl
+julia> LOAD_PATH
+3-element Vector{String}:
+ "@"
+ "@v#.#"
+ "@stdlib"
+```
+
+Briefly speaking, Julia will try to load package `BenchmarkTools` in `@` environment, which is the current activated project.
+If not found then try the version-specific global environment `@v#.#`, which is stored in `~/.julia/environments/v1.6` for Julia 1.6.
+In the previous case, Julia found `BenchmarkTools` in `@v#.#`, i.e., `~/.julia/environments/v1.6/Project.toml`,
+thus it successfully loads the package.
+
+A convenient command to create custom shared environment is to prefix the name with `@`:
+
+```julia-repl
+(@v1.6) pkg> activate @mytoolbox # require Julia 1.4 or later
+  Activating new project at `~/.julia/environments/mytoolbox`
+```
+
+This provides a quick way to activate the project environment stored in `~/.julia/environments` folder.
+
+### Shared environment caveat: the reproducibility issue
+
+Using shared environment can cause reproducibility issues. For instance,
+
+```julia
+(MyProject) pkg> generate MyPackage
+  Generating  project MyPackage:
+    MyPackage/Project.toml
+    MyPackage/src/MyPackage.jl
+
+# ... some modifications to MyPackage/src/MyPackage.jl
+
+julia> print(read("MyPackage/Project.toml", String))
+name = "MyPackage"
+uuid = "c10752ec-72c1-4eb2-a399-580941ff7382"
+version = "0.1.0"
+
+julia> print(read("MyPackage/src/MyPackage.jl", String))
+module MyPackage
+
+using BenchmarkTools
+
+greet() = print("Hello World!")
+
+end # module
+```
+
+Now when you tries to load `MyPackage`, because `BenchmarkTools` is recorded in the global shared environment `v#.#`,
+Julia still succeeds loading it but will print a warning message:
+
+```julia
+julia> using MyPackage
+┌ Warning: Package MyPackage does not have BenchmarkTools in its dependencies:
+│ - If you have MyPackage checked out for development and have
+│   added BenchmarkTools as a dependency but haven't updated your primary
+│   environment's manifest file, try `Pkg.resolve()`.
+│ - Otherwise you may need to report an issue with MyPackage
+└ Loading BenchmarkTools into MyPackage from project dependency, future warnings for MyPackage are suppressed.
+```
+
+Imaging that you want to ship your wonderful package to other people, because `BenchmarkTools` is not recoreded in `MyPackage/Project.toml`,
+you can't ensure that `BenchmarkTools` can be successfully loaded even if they initialize your package with `pkg> instantiate`.
+In such sense, this is a reproducibility warning; when you develop a package that meant to be shared by others, you should avoid doing this.
```

Comment:
```suggestion
Imagine that you want to ship your wonderful package to other people. Because `BenchmarkTools` is not recorded in `MyPackage/Project.toml`,
you can't ensure that `BenchmarkTools` will be successfully loaded. For this reason, developers of shared packages should be especially mindful of reproducibility issues when using shared environments.
```